### ISSUE TITLE
Black squared corners showing up behind rounded corners

### ISSUE THREAD

**Issue by asteraceae** [yshui/picom#808]
### Platform
Debian 11 bullseye

### GPU, drivers, and screen setup
nouveau drivers,  in the midst of setting up nvidia drivers

### Environment
i3-gaps + xfce4-panel

### Current Behavior
![image](https://i.ibb.co/3sZxTfk/wrong.png)

Can not figure out for the life of me why this is happening.  Shadows are set to false. This does not happen to all windows, thunar and some dropdown menus are rendered fine.  However this is present on the majority of windows.

**Comment by asteraceae** [comment]
setting up nvidia drivers did not help

**Comment by Wolfdesroyer8** [comment]
Can you show me your picom config?

**Comment by 26hz** [comment]
I have the same issues my config :
corner-radius = 20.0;
round-borders = 4
rounded-corners-exclude = [
  "class_g = 'Polybar'",
];
shadow-radius = 0;

**Comment by Wolfdesroyer8** [comment]
How are you running picom because if you dont specify the backend in the config then you will get black corners when you run with `picom --experimental-backends` so you could try adding `backend = "glx";`

tl;dr try running picom like this `picom` or this `picom --experimental-backends` and if that does fix it then try adding `backend = glx;` to your config.

**Comment by c02y** [comment]
> How are you running picom because if you dont specify the backend in the config then you will get black corners when you run with `picom --experimental-backends` so you could try adding `backend = "glx";`
> 
> tl;dr try running picom like this `picom` or this `picom --experimental-backends` and if that does fix it then try adding `backend = glx;` to your config.

I'm using both `picom --experimental-backends` and `backend = "glx";`, it doesn't help.

**Comment by appuchias** [comment]
I'm having the same problem but with white squares.
![image](https://user-images.githubusercontent.com/46938448/175793431-79222206-0f3b-4c54-b9a3-49fcbcb4b084.png)
![image](https://user-images.githubusercontent.com/46938448/175793505-0fc18c2d-d259-4c7d-b7ea-7155969705d1.png)

I'm using AwesomeWM

Here is my picom.conf, I'm running `picom --experimental-backends -b` on startup. (It's renamed to be able to upload it, I have it as `picom.conf` in my system)
[picom.conf](https://github.com/yshui/picom/files/8985724/picom.txt)

**Comment by ElnuDev** [comment]
> How are you running picom because if you dont specify the backend in the config then you will get black corners when you run with `picom --experimental-backends` so you could try adding `backend = "glx";`
> 
> tl;dr try running picom like this `picom` or this `picom --experimental-backends` and if that does fix it then try adding `backend = glx;` to your config.

Switching to `backend = "glx";` actually fixed the issue for me. I'm using a Nvidia GPU.

**Comment by elamperti** [comment]
This happens for me with both xrender and glx, tried forcing backends with both glx and xrender, and even both with `--legacy-backends`. [This](https://pastebin.com/sQ29jsuY) is my config, I'm seeing this with the [`WhiteSur-Dark` theme](https://github.com/vinceliuice/WhiteSur-gtk-theme) on popups/context menus. Using i3wm, picom v10.2 (b700a).

**Comment by absolutelynothelix** [comment]
it should be fine for all kinds of configurations now.
1) try the latest commit;
2) if the latest commit doesn't fix the issue for you, apply the #1003 on top of it;
3) if applying the #1003 on top of the latest commit doesn't fix the issue for you, apply the #1002 as well.

(all the mentioned pull requests are going to be merged eventually)

if the issue isn't gone after doing all the steps, report back.

**Comment by absolutelynothelix** [comment]
all the pull requests mentioned above were merged into the `next` branch and included in the v11-rc1 pre-release.

closing as stale.
